apps/heft/src/cli/actions/BuildAction.ts

```diff
@@ -74,17 +74,69 @@ export interface ICopyStaticAssetsConfiguration extends ISharedCopyStaticAssetsC
   destinationFolderNames: string[];
 }
 
+/**
+ * @public
+ */
+export interface IEmitModuleKindBase<TModuleKind> {
+  moduleKind: TModuleKind;
+  outFolderPath: string;
+}
+
+/**
+ * @public
+ */
+export type IEmitModuleKind = IEmitModuleKindBase<
+  'commonjs' | 'amd' | 'umd' | 'system' | 'es2015' | 'esnext'
+>;
+
+/**
+ * @public
+ */
+export type CopyFromCacheMode = 'hardlink' | 'copy';
+
+/**
+ * @public
+ */
+export interface ISharedTypescriptConfiguration {
+  /**
+   * Can be set to 'copy' or 'hardlink'. If set to 'copy', copy files from cache. If set to 'hardlink', files will be
+   * hardlinked to the cache location. This option is useful when producing a tarball of build output as TAR files
+   * don't handle these hardlinks correctly. 'hardlink' is the default behavior.
+   */
+  copyFromCacheMode?: CopyFromCacheMode | undefined;
+
+  /**
+   * If provided, emit these module kinds in addition to the modules specified in the tsconfig.
+   * Note that this option only applies to the main tsconfig.json configuration.
+   */
+  additionalModuleKindsToEmit?: IEmitModuleKind[] | undefined;
+}
+
+/**
+ * @public
+ */
+export interface ITypescriptConfiguration extends ISharedTypescriptConfiguration {
+  tsconfigPaths: string[];
+  tslintConfigPath: string | undefined;
```

Comment:
If the path is configurable, how will the VS Code extension find it?

Reply:
Hmm that's a good point. Perhaps we shouldn't bother making it configurable.